lib/statesman/adapters/active_record.rb

```diff
@@ -128,10 +128,27 @@ def serialized?(transition_class)
       end
 
       def transition_conflict_error?(err)
-        err.message.include?(@transition_class.table_name) &&
+        return true if unique_indexes.any? { |i| err.message.include?(i.name) }
+
+        err.message.include?(transition_class.table_name) &&
           (err.message.include?("sort_key") || err.message.include?("most_recent"))
       end
 
+      def unique_indexes
+        ::ActiveRecord::Base.connection.
+          indexes(transition_class.table_name).
+          select do |index|
+            next unless index.unique
+
+            index.columns.sort == [parent_join_foreign_key, "sort_key"] ||
+              index.columns.sort == [parent_join_foreign_key, "most_recent"]
+          end
+      end
+
+      def parent_join_foreign_key
```

Comment:
Do we have any guarantee that people will name their columns like this? Seems like a more solid assumption than the one you've removed in this PR, but was wondering if we should do something smarter for this too?

Reply:
Good shout. Found a smarter way to do it.